### PR TITLE
[TT-4858] Implement OAS track / do not track endpoint

### PR DESCRIPTION
**Pull request by titpetric**
https://tyktech.atlassian.net/browse/TT-4858

### CODE REVIEW

**Comment by buger** [comment]
**API tests result - postgres15-sha256 env: success** :white_check_mark: 
Branch used: refs/pull/5948/merge
Commit:  
Triggered by: pull_request (@titpetric)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/7554576627)

**Comment by buger** [comment]
**API tests result - mongo44-sha256 env: success** :white_check_mark: 
Branch used: refs/pull/5948/merge
Commit:  
Triggered by: pull_request (@titpetric)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/7554576627)

**Comment by buger** [comment]
**API tests result - postgres15-murmur64 env: success** :white_check_mark: 
Branch used: refs/pull/5948/merge
Commit:  
Triggered by: pull_request (@titpetric)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/7554576627)

**Comment by buger** [comment]
**API tests result - mongo44-murmur64 env: success** :white_check_mark: 
Branch used: refs/pull/5948/merge
Commit:  
Triggered by: pull_request (@titpetric)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/7554576627)

File: apidef/oas/operation.go [hunk 1]
```diff
@@ -65,6 +65,12 @@ type Operation struct {
 
 	// CircuitBreaker contains the configuration for the circuit breaker functionality.
 	CircuitBreaker *CircuitBreaker `bson:"circuitBreaker,omitempty" json:"circuitBreaker,omitempty"`
+
+	// TrackEndpoint contains the configuration for enabling analytics and logs.
```
Comment (jeffy-mathew) [inline apidef/oas/operation.go:69]: shall we add a the location of this mw in TykClassic Def

File: apidef/oas/track_endpoint.go [hunk 2]
```diff
@@ -0,0 +1,75 @@
+package oas
+
+import (
+	"github.com/TykTechnologies/tyk/apidef"
+)
+
+// TrackEndpoint configures Track or DoNotTrack behaviour for an endpoint.
+// Tyk classic API definition: `version_data.versions[].extended_paths.(track_endpoints|do_not_track_endpoints)`.
```
Comment (titpetric) [inline apidef/oas/track_endpoint.go:8]:
```suggestion
// Tyk classic API definition: `version_data.versions..extended_paths.track_endpoints`, `version_data.versions..extended_paths.do_not_track_endpoints`.
```

**Pull request merged**